proto-lens/src/Data/ProtoLens/GRPC/Types.hs

```diff
@@ -0,0 +1,21 @@
+{-# LANGUAGE GADTs #-}
+
+module Data.ProtoLens.GRPC.Types where
+
+import Data.ProtoLens.Message (Message)
+
+
+type ServerHandler       a b = a                                 -> IO b
+type ServerReaderHandler a b =      IO (Maybe a)                 -> IO b
+type ServerWriterHandler a b = a                 -> (b -> IO ()) -> IO ()
+type ServerRWHandler     a b = a -> IO (Maybe a) -> (b -> IO ()) -> IO ()
+
+data SomeHandler where
+  NormalHandler :: (Message a, Message b) => ServerHandler       a b -> SomeHandler
+  ReaderHandler :: (Message a, Message b) => ServerReaderHandler a b -> SomeHandler
+  WriterHandler :: (Message a, Message b) => ServerWriterHandler a b -> SomeHandler
+  BiDiHandler   :: (Message a, Message b) => ServerRWHandler     a b -> SomeHandler
+
+class Service s where
+  packHandlers :: s -> [SomeHandler]
```

Comment:
Can this also support client-side RPC calls?  Or would we need another method?

Reply:
there are client-side rpcs?

Reply:
I mean for client-side code that sends requests to gRPC servers.  The `packHandlers` method seems more suited for implementing the gRPC servers themselves.

Reply:
good question. do you have a mental model of what the client-side interface should look like? in particular the streaming interfaces.